engine/controller/engine_controller.py

```diff
@@ -70,22 +69,38 @@ def get_engine_meta(self, task: Task) -> ReturnDict:
         return ReturnDict(status=StatusCode.OK, content=engine.get_engine_meta())
 
     @engine_controller_exception_handler
-    def append_frame(
-        self, task: Task, frame_component_raw: FrameModel, force=False
-    ) -> ReturnList:
+    def append_empty_frame(self, task: Task, to_chapter: str):
         """
-        append frame: Frame into game content
+        append an empty frame according to chapter
 
-        @return: metadata for engine
+        @param to_chapter: append to which chapter
+        @param task:
+        @return: the added frame id
+
+        """
+        pass
+
+    @engine_controller_exception_handler
+    def modify_frame(
```

Comment:
now, the commit function merged with this one, it will call commit and save all the changes when call it